src/streamlink/plugins/app17.py

```diff
@@ -0,0 +1,68 @@
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import http, useragents
+from streamlink.stream import HLSStream, RTMPStream
+
+API_URL = "https://api-dsa.17app.co/api/v1/liveStreams/isUserOnLiveStream"
+ROOM_URL = "http://17app.co/share/live/{0}"
+
+_url_re = re.compile(r"http://17app.co/share/(?P<page>[^/]+)/(?P<channel>[^/&?]+)")
+_userid_re = re.compile(r'"userID"\s*:\s*"(.+?)"')
+_rid_re = re.compile(r'"liveStreamID"\s*:\s*(\d+)')
+_status_re = re.compile(r'"userIsOnLive"\s*:\s*([A-z]+)')
+_rtmp_re = re.compile(r'"url"\s*:\s*"(.+?)"')
+
+
+class App17(Plugin):
+    @classmethod
+    def can_handle_url(cls, url):
+        return _url_re.match(url)
+
+    def _get_streams(self):
+        match = _url_re.match(self.url)
+        page = match.group("page")
+
+        http.headers.update({'User-Agent': useragents.CHROME})
+
+        if page == 'user':
+            res = http.get(self.url)
+            userid = _userid_re.search(res.text).group(1)
+            data = {
+                "targetUserID": userid
+            }
+            api = http.post(API_URL, data=data)
```

Comment:
As this is returning JSON you could get rid of some of the error prone regex stuff and add a `validate.Schema` to parse the JSON.
```python
_user_api_schema = validate.Schema(validate.all(
    {"data": validate.transform(parse_json)},
    validate.get("data"),
    {
        "liveStreamID": int,
        "userIsOnLive": int,

    }
))
```

And instead of doing the regex you just do:
```python
api = http.post(API_URL, data={"targetUserID": userid})
data = http.json(api, schema=_user_api_schema)

if data["liveStreamID"] == 0:
    self.logger.info("Stream currently unavailable.")
    return
```

__This is only a suggestion, you don't have to change this.__